ggshield/core/git_shell.py

```diff
@@ -131,16 +131,23 @@ def is_valid_git_commit_ref(ref: str) -> bool:
     return True
 
 
-def get_list_commit_SHA(commit_range: str) -> List[str]:
+def get_list_commit_SHA(
+    commit_range: str, max_count: Optional[int] = None
+) -> List[str]:
     """
     Retrieve the list of commit SHA from a range.
     :param commit_range: A range of commits (ORIGIN...HEAD)
+    :param max_count: If set, limits the number of SHA returned to this amount
     """
+
+    cmd = [GIT_PATH, "rev-list", "--reverse", *commit_range.split()]
```

Comment:
I know the PR is not in review yet, but a quick remark : given the format of `commit_range`, I don't understand the interest of `*commit_range.split()` instead of simply `commit_range` ?  
I think this is a legacy piece of code introduced in [this PR](https://github.com/GitGuardian/ggshield/pull/10).

Reply:
This is required because there is at least one case where `commit_range` still contains a space: https://github.com/GitGuardian/ggshield/blob/jchaumont-agateau/303/smarter-prepush/ggshield/cmd/secret/scan/prereceive.py#L130.